sdk/metrics-testing/src/main/java/io/opentelemetry/sdk/testing/assertj/metrics/MetricDataAssert.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.testing.assertj.metrics;
+
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricDataType;
+import io.opentelemetry.sdk.resources.Resource;
+import org.assertj.core.api.AbstractAssert;
+
+/** Test assertions for {@link MetricData}. */
+public class MetricDataAssert extends AbstractAssert<MetricDataAssert, MetricData> {
+  protected MetricDataAssert(MetricData actual) {
+    super(actual, MetricDataAssert.class);
+  }
+
+  /** Ensures the {@link Resource} associated with a metric matches the expected value. */
+  public MetricDataAssert hasResource(Resource resource) {
```

Comment:
What do you think about just `hasResourceAttributes` and `hasResourceAttributesSatisfying`? Resource only has one field, attributes

Reply:
While I half-agree, in practice almost all the tests (and code) just pass `Resource` wholesale, and check for the one static instance that was given to an SDK.  Very rarely are folks checking resource detection/attribute scenarios in Metrics code.

I imagine you could add those two methods, but I don't think I'd wind up using them for most of the metrics tests.

Reply:
just FYI, Resource also has `schemaUrl` now; not sure if that alters the fundamentals, though.